piicatcher/generators.py

```diff
@@ -44,9 +42,13 @@ def column_generator(
 
 
 def _get_table_count(
-    schema: CatSchema, table: CatTable, dbinfo: Type[DbInfo], connection
+    schema: CatSchema,
+    table: CatTable,
+    dbinfo: Type[DbInfo],
+    connection,
+    source=CatSource,
 ) -> int:
-    count = dbinfo.get_count_query(schema.name, table.name)
+    count = dbinfo.get_count_query(schema.name, table.name, source.project_id)
```

Comment:
This will still cause problems if the next connector you add requires a _different_ parameter to be passed.

You need to pass the `project_id` to the BigQuery constructor through the `get_dbinfo` function. But for the `get_count_query` method to be able to access instance properties you have to change that method from a class method to an instance method.